core/iwasm/libraries/lib-socket/inc/wasi_socket_ext.h

```diff
@@ -85,6 +85,20 @@ typedef struct __wasi_addr_info_hints_t {
     uint8_t hints_enabled;
 } __wasi_addr_info_hints_t;
 
+typedef struct __wasi_linger {
+    uint8_t l_onoff;
+    uint32_t l_linger_s;
+} __wasi_linger;
+
+typedef struct __wasi_in_addr_t {
```

Comment:
could we use `__wasi_addr_ip4_t` instead, as this is a structure that we use for representing IPv4 addresses?